src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -241,6 +241,73 @@ export default class Fms {
         return routeSegments.join(DIRECT_ROUTE_SEGMENT_SEPARATOR);
     }
 
+    /**
+     * Return the next waypoint which has an altitude restriction
+     *
+     * @for Fms
+     * @property nextAltitudeRestrictedWaypoint
+     * @type {WaypointModel}
+     */
+    get nextAltitudeRestrictedWaypoint() {
+        const waypoints = this.getAltitudeRestrictedWaypoints();
+
+        return waypoints[0];
+    }
+
+    /**
+     * Return the next waypoint which has an "AT" altitude restriction
+     *
+     * @for Fms
+     * @property nextHardAltitudeRestrictedWaypoint
+     * @type {WaypointModel}
+     */
+    get nextHardAltitudeRestrictedWaypoint() {
+        const waypoints = this.getAltitudeRestrictedWaypoints()
+            .filter((waypoint) => waypoint.altitudeMaximum === waypoint.altitudeMinimum);
+
+        return waypoints[0];
+    }
+
+    /**
+     * Return the next waypoint which has an "AT" speed restriction
+     *
+     * @for Fms
+     * @property nextHardSpeedRestrictedWaypoint
+     * @type {WaypointModel}
+     */
+    get nextHardSpeedRestrictedWaypoint() {
```

Comment:
I'm very much torn between getters and methods for these next few.  It could be argued either way, and both argument I'd say has merit, but I'm leaning towards methods for these getters.  What do you think?